website/blog/2024-03-07-using-inner-blocks.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Using inner blocks
+description: Various tips and useful features for your project included in Eightshift DevKit.
+slug: using-inner-blocks
+authors: obradovic
+date: 2024-03-07
+tags: [eightshift, boilerplate, inner blocks]
+hide_table_of_contents: false
+---
+
+InnerBlocks in WordPress enable the creation of nested block structures within the Gutenberg editor, allowing for the development of more complex and customizable content layouts.
+<!--truncate-->
+
+## Why use inner blocks?
+
+Use of inner blocks allows for more flexibility when building content. Let’s say you’re building a simple two-column layout with image on one side and content on the other side. You can define the components that will be used (e.g. header, paragraph, list, button) and then just toggle the ones you don’t need.
+
+This approach works perfectly fine when you have a strict layout and don’t want larger content structure variations. Using inner blocks may introduce a bit more complexity - both for developers and content editors, but ultimately gives the editors much more freedom.
+
+Another example where inner blocks are used is the Carousel block. Using inner blocks allows you to add as many slides as you like in your carousel. Let’s not forget about Columns and Group blocks. These also use inner blocks.
+
+## Adding inner block support
+
+The example used in this blog post is a two-column Card that was made for one project. First thing that needs to be updated is the ***manifest.json*** file. For the sake of simplicity and not having too much code, the parts that aren’t related to the inner blocks have been removed.
+
+```json
+{
+	"hasInnerBlocks": true,
+	"innerBlocksDependency": [
+		"heading",
+		"paragraph"
+	],
+	"attributes": {
+		"cardTwoColumnAllowedBlocks": {
+			"type": "array",
+			"items": {
+				"type": "string"
+			},
+			"default": [
+				"eightshift-boilerplate/paragraph",
+				"eightshift-boilerplate/heading",
+				"eightshift-boilerplate/button",
+				"eightshift-boilerplate/lists"
+			]
+		}
+	}
+}
+```
+
+The `hasInnerBlocks` key defines that this block supports inner blocks. The `innerBlocksDependency` is not mandatory, but it can be used to add inner blocks automatically. You can see an example of using it in the Columns block `editor.js` file.
+
+In case we need to limit the use of inner blocks, we can do so by defining the allowed blocks inside attributes. The attribute name should be in the format `{blockName}AllowedBlocks`. To define the allowed blocks, simply add the full block names in the `default` key array, as shown in the example.
```

Comment:
```suggestion
In case we need to limit which blocks can be added inside inner blocks, we can do so with an attribute. Usually, the attribute is named `{blockName}AllowedBlocks`. Set the default value as an array of strings that represent block names you want to allow.

Please note that the block names must include the block namespace! (`eightshift-boilerplate` by default, check in Global manifest)
```